### PR TITLE
[WFCORE-2578] Add ability to configure bind addresses for outbound XNIO connections

### PR DESCRIPTION
**Pull request by dmlloyd**
https://issues.jboss.org/browse/WFCORE-2578
https://issues.jboss.org/browse/WFCORE-2610
https://issues.jboss.org/browse/WFCORE-2611
https://issues.jboss.org/browse/WFCORE-2612

### CODE REVIEW

**Comment by dmlloyd** [comment]
I'm not sure why the commits are sorted like this in the UI, but the actual order from oldest to newest is:
* fddd07abf7060242e0f351533b54650529ea93ff
* a7079db70cc0bd598c53f0b1a39a58a73f871f90
* 2aad5e8999e883b23b5820ce3b5502f569c424b4
* 1a6ad24e8693b45bab4906de6e35955c785cc44c

**Comment by dmlloyd** [comment]
This change was reviewed previously by @ctomc.

File: io/subsystem/src/main/java/org/wildfly/extension/io/OutboundBindAddressUtils.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+ * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
+ */
+final class OutboundBindAddressUtils {
+    private OutboundBindAddressUtils() {
+    }
+
+    static InetSocketAddress getBindAddress(final ModelNode operation, final OperationContext context) throws OperationFailedException {
+        return new InetSocketAddress(Inet.parseInetAddress(
+            OutboundBindAddressResourceDefinition.BIND_ADDRESS.resolveModelAttribute(context, operation).asString()),
+            OutboundBindAddressResourceDefinition.BIND_PORT.resolveModelAttribute(context, operation).asInt()
```
Comment (kabir) [inline io/subsystem/src/main/java/org/wildfly/extension/io/OutboundBindAddressUtils.java]: This attribute is not required, so it probably needs a default or something like that

**Comment by dmlloyd** [comment]
OK all comments should be addressed, hopefully.

**Comment by bstansberry** [comment]
I opened #2305 with fixes for this.

**Pull request merged**